packages/dhis2/src/http.js

```diff
@@ -0,0 +1,228 @@
+import { expandReferences, encode } from '@openfn/language-common/util';
+import { generateUrl, handleResponse } from './Utils';
+import * as client from './Client';
+
+/**
+ * Options object
+ * @typedef {Object} RequestOptions
```

Comment:
Worried about these docs but they're not technically part of this PR. Let me think about it. Referenced here: https://github.com/OpenFn/adaptors/issues/975#issuecomment-2711368315